src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -23,10 +23,16 @@
  */
 public class TopicManagementService implements Service {
   private final MultiClusterTopicManagementService _multiClusterTopicManagementService;
+  private static final CompletableFuture<Void> TOPIC_PARTITION_READY = new CompletableFuture<>();
 
   public TopicManagementService(Map<String, Object> props, String serviceName) throws Exception {
     Map<String, Object> serviceProps = createMultiClusterTopicManagementServiceProps(props, serviceName);
     _multiClusterTopicManagementService = new MultiClusterTopicManagementService(serviceProps, serviceName);
+    TOPIC_PARTITION_READY.complete(null);
+  }
+
+  public static CompletableFuture<Void> topicPartitionReady() {
```

Comment:
Static methods are going to kill any kind of testability which is why there's a suggestion to work on unit testing.  Then it will become more clear what's safe to make static.

Reply:
Non-static usages now. Unit testing is in the backlog.